server/src/main/java/io/crate/analyze/relations/UnionSelect.java

```diff
@@ -109,4 +102,42 @@ public String toString() {
     public boolean isDistinct() {
         return isDistinct;
     }
+
+    private List<ScopedSymbol> unionizeOutputs(List<Symbol> left, List<Symbol> right) {
+        ArrayList<ScopedSymbol> outputs = new ArrayList<>(left.size());
+        for (int i = 0; i < left.size(); i++) {
+            outputs.add(unionizeSymbol(left.get(i), right.get(i)));
+        }
+        return outputs;
+    }
+
+    private ScopedSymbol unionizeSymbol(Symbol left, Symbol right) {
+        return new ScopedSymbol(name, Symbols.pathFromSymbol(left), unionizeTypes(left.valueType(), right.valueType()));
+    }
```

Comment:
I'd keep this inlined in the ctor. It's not re-used and hardly worth breaking out.